ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/PrivateTransactionWithMetadata.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.privacy;
+
+import org.hyperledger.besu.ethereum.privacy.storage.PrivateTransactionMetadata;
+import org.hyperledger.besu.ethereum.rlp.RLPException;
+import org.hyperledger.besu.ethereum.rlp.RLPInput;
+import org.hyperledger.besu.ethereum.rlp.RLPOutput;
+
+import java.util.Objects;
+
+public class PrivateTransactionWithMetadata {
+  private final PrivateTransaction privateTransaction;
+  private final PrivateTransactionMetadata privateTransactionMetadata;
+
+  public static PrivateTransactionWithMetadata readFrom(final RLPInput input) throws RLPException {
+    input.enterList();
+    final PrivateTransaction privateTransaction = PrivateTransaction.readFrom(input.readAsRlp());
+    final PrivateTransactionMetadata privateTransactionMetadata =
+        PrivateTransactionMetadata.readFrom(input.readAsRlp());
+    input.leaveList();
+    return new PrivateTransactionWithMetadata(privateTransaction, privateTransactionMetadata);
+  }
+
+  public PrivateTransactionWithMetadata(
+      final PrivateTransaction privateTransaction,
```

Comment:
Don't we need the VersionedPrivateTransaction here? Rehydration seems to work, but I thought that the version should be necessary so we do not execute transactions with the wrong version ...
Or do private transactions that were sent with the wrong version not make their way into this list?